src/components/Tooltip/Tooltip.tsx

```diff
@@ -106,10 +116,10 @@ const Tooltip = ({
       clearTimeout(hideTooltipTimer.current);
     }
 
-    showTooltipTimer.current = setTimeout(
-      () => setVisible(true),
-      enterTouchDelay
-    ) as unknown as NodeJS.Timeout;
+    showTooltipTimer.current = setTimeout(() => {
+      setTouched(true);
```

Comment:
Adding another setState under this setTimeout will cause another rerender for React < 18.00 due to how the automatic batching works. 
Performance wise I don't think it would be a problem but this could be easily fixed with a ref, what do you think?

We can replace `const [touched, setTouched] = React.useState(false);` with `const touched = useRef(false)` and then:

```suggestion
      touched.current = true
```

Reply:
Will investigate that, thanks for hint!

Reply:
For reference: https://reactjs.org/blog/2022/03/29/react-v18.html#new-feature-automatic-batching

Reply:
Done, thanks!